packages/itwinui-react/src/core/Table/columns/actionColumn.tsx

```diff
@@ -94,45 +102,41 @@ export const ActionColumn = <T extends Record<string, unknown>>({
               });
             };
             return (
-              <MenuItem
+              <Checkbox
                 key={column.id}
-                startIcon={
-                  <Checkbox
-                    checked={checked}
-                    disabled={column.disableToggleVisibility}
-                    onClick={(e) => e.stopPropagation()}
-                    onChange={onClick}
-                    aria-labelledby={`iui-column-${column.id}`}
-                  />
-                }
-                onClick={onClick}
+                checked={checked}
                 disabled={column.disableToggleVisibility}
-              >
-                <div id={`iui-column-${column.id}`}>
-                  {column.render('Header')}
-                </div>
-              </MenuItem>
+                onClick={(e) => e.stopPropagation()}
+                onChange={onClick}
+                label={column.render('Header')}
+              />
             );
           });
 
-      const dropdownMenuProps =
+      const popoverProps =
         typeof columnManager !== 'boolean'
           ? columnManager.dropdownMenuProps
           : {};
 
       return (
-        <DropdownMenu
-          {...dropdownMenuProps}
-          menuItems={headerCheckBoxes}
+        <Popover
+          content={
+            <Surface as={'fieldset'}>
+              <Flex flexDirection='column' alignItems='flex-start'>
+                {headerCheckBoxes()}
+              </Flex>
+            </Surface>
+          }
+          {...popoverProps}
           onVisibleChange={(open) => {
             setIsOpen(open);
-            dropdownMenuProps?.onVisibleChange?.(open);
+            popoverProps?.onVisibleChange?.(open);
           }}
         >
           <IconButton styleType='borderless' isActive={isOpen} ref={buttonRef}>
```

Comment:
We should add `label` here. And we don't need to set the `isActive` prop, since `Popover` now provides popover-open styling (#2076).

```jsx
<IconButton
  styleType='borderless'
  ref={buttonRef}
  label='Column manager'
>
```